MeshLib/IO/readMeshFromFile.cpp

```diff
@@ -45,9 +45,36 @@ namespace IO
 MeshLib::Mesh* readMeshFromFile(const std::string &file_name)
 {
 #ifdef USE_PETSC
-    MeshLib::IO::NodePartitionedMeshReader read_pmesh(PETSC_COMM_WORLD);
-    const std::string file_name_base = BaseLib::dropFileExtension(file_name);
-    return read_pmesh.read(file_name_base);
+    int world_size;
+    MPI_Comm_size(MPI_COMM_WORLD, &world_size);
+    if (world_size > 1)
+    {
+        MeshLib::IO::NodePartitionedMeshReader read_pmesh(PETSC_COMM_WORLD);
+        const std::string file_name_base = BaseLib::dropFileExtension(file_name);
+        return read_pmesh.read(file_name_base);
+    }
+    else if (world_size == 1)
+    {
+        MeshLib::Mesh* mesh = nullptr;
```

Comment:
there is MeshLib::IO::readMeshFromFile() function already capturing msh/vtk stuff.